frontend/test/metabase/scenarios/dashboard/text-parameters.cy.spec.js

```diff
@@ -0,0 +1,79 @@
+import {
+  restore,
+  editDashboard,
+  saveDashboard,
+  visitDashboard,
+  setFilter,
+  filterWidget,
+  addTextBox,
+  popover,
+} from "__support__/e2e/helpers";
+
+describe("scenarios > dashboard > parameters in text cards", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.createDashboard().then(({ body: { id: DASHBOARD_ID } }) => {
+      visitDashboard(DASHBOARD_ID);
+    });
+  });
+
+  it("should show instructional text for text cards with no variables", () => {
+    addTextBox("Text card with no variables", {
+      parseSpecialCharSequences: false,
+    });
+    editDashboard();
+    setFilter("Number", "Equal to");
+    cy.findByText(
+      "You can connect widgets to {{variables}} in text cards.",
+    ).should("exist");
+    cy.icon("info").should("exist");
+  });
+
+  it("should allow dashboard filters to be connected to tags in text cards", () => {
+    addTextBox("Variable: {{foo}}", { parseSpecialCharSequences: false });
+    editDashboard();
+    setFilter("Number", "Equal to");
+
+    cy.findByText("Select…").click();
+    cy.findByText("foo").click();
+    saveDashboard();
+
+    filterWidget().click();
+    cy.findByPlaceholderText("Enter a number").type(`1{enter}`);
+    cy.button("Add filter").click();
+    cy.findByText("Variable: 1").should("exist");
+
+    cy.findByText("1").click();
+    popover().within(() => {
+      cy.findByRole("textbox").click().type("2{enter}");
+      cy.button("Update filter").click();
+    });
+    cy.findByText("Variable: 1 and 2").should("exist");
+  });
+
+  it("should translate parameter values into the instance language", () => {
+    cy.request("GET", "/api/user/current").then(({ body: { id: USER_ID } }) => {
+      cy.request("PUT", `/api/user/${USER_ID}`, { locale: "en" });
+    });
+    cy.request("PUT", `/api/setting/site-locale`, { value: "fr" });
+    cy.reload();
```

Comment:
Not sure I fully understand these 3-4 commands.

Why do we have to explicitly set user locale to `en`?
Why do we have to reload?

Reply:
I suppose we don't need to explicitly set user locale to `en`. I'll remove it.

The reload is because the translation bundles (for both site and user locales) are provided during the initial page load as part of the `index.html` template. So just setting the site locale over the API doesn't make the translations available yet -- we need a fresh page load as well.

Reply:
Actually, I remember now why I set the user locale to `en` explicitly. It's because the default user locale is to match the site locale. So if we just change the site locale, the user locale will also change. But here, we intentionally want them to diverge for the sake of the test.

I'll add a comment to clarify that.